tutorials/marble/version-control.md

```diff
@@ -0,0 +1,393 @@
+
+
+![Git Push Credentials](images/version-control/git-push-credentials.png)
+
+If successful you will see a `Successfully pushed` status pop up in the lower right corner.
+
+![Push Successful](images/version-control/push-successful.png)
+
+### Creating a Branch
+
+Click the down arrow in the `Current Branch` section in the Git sidebar.  It will reveal the `Branches` window and show all the created branches.
+
+Click the `New Branch` button.
+
+![New Branch Button](images/version-control/new-branch-button.png)
+
+Fill out the form and input a branch name, and click which branch to base the new branch off of.  After that click `Create Branch`.
+
+It is good practice to not use spaces or special characters in the branch name other than dashes.
+
+
+![New Branch Form](images/version-control/new-branch-dialog.png)
+
+Git will automatically switch you into the new branch.  Your newly created branch will be shown under `Current Branch` and be in the list of branches.  
+
+![Create Branch Successful](images/version-control/create-branch-successful.png)
+
+### Stashing Files
+
+Stashing a file means to temporarily store files you're working on so the changes made to them will not affect other files, 
+and changes to other files will not affect the working file.  Stashing is typically done before switching branches, so 
+the changes from other branches will not overwrite changes in files you are currently working on.
+
+Currently there is no mechanism in the GUI to stash files.  Files can only be stashed from the Terminal.
+
+[Stash Files from Terminal](#stash-terminal)
+
+Once the command is run all files listed under `Changed` will be moved to the `Stash` and will not be seen. 
+
+
+### Switching Branches
+
+```{note}
+You can only switch branches when you have no files staged or changed.
+ 
+To switch branches simply click another branch listed under the `Branches` section.
+```
+
+```{note}
+You can only switch to another local branch.  Branches with the `origin` prefix are remote branches and cannot be switched into.
+``` 
+
+
+### Using Git with the Terminal
+
+### Set Up Your Git Identifier
+
+If you are working in a team this will show who made a certain commit.
+
+```
+git config --global user.name "Your Name"
+git config --global user.email "email@email.com"
+```
+
+### Staging a File
+
+Make sure you are in the same directory as the file you want to stage.
+
+```
+git add <filename>
+```
+
+### View Staged Files
+
+If you have multiple files staged and ready to commit you can see the list by using `git status`.
+
+```
+git status
+```
+
+
+### Making a Commit
+
+Add a brief descriptive message about your commit.
+
+```
+git commit -m "your brief commit message goes here"
+```
+
+### Pushing a Commit
+
+After your commit is complete, push it to the remote repository
+
+```
+git push
+```
+
+### Stashing Files
+
+Make sure you are in the same directory as the file you want to stash.  
```

Comment:
Not necessary, you can stash from anywhere in the repo.

Reply:
Same thing with the Staging part.I'll add an example with a folder path